.travis.yml

```diff
@@ -101,210 +76,197 @@ _helpers:
+# key is only needed to provide a container for
+# the &osx_env_<version> YAML anchor).
+- &osx_env_py35
+  - PYTHON_VERSION: "3.5"
   - *env_pyenv
   - *env_path
-- &osx_pypi_deploy_base_1010
-  <<: *osx_pypi_deploy_base_1011
-  osx_image: xcode6.4
-  env:
-  - &env_os1010_msg Build and deploy to PYPI of OS X 10.10 binary wheel=
-  - *env_py36
+- &osx_env_py36
+  - PYTHON_VERSION: "3.6"
+  - *env_pyenv
+  - *env_path
+- &osx_env_py37
+  - PYTHON_VERSION: "3.7"
   - *env_pyenv
   - *env_path
 
+- &linux_pypi_deploy_base
+  <<: *_generic_deploy_base
+  <<: *mainstream_python_base
+  stage: &deploy_stage_name Linux build and deploy
+  deploy:
+    <<: *_deploy_defaults
+    # `skip_cleanup: true` is required to preserve binary wheels, built
+    # inside of manylinux1 docker container during `script` step above.
+    skip_cleanup: true
+
+
 os: linux
 
-jobs:
+matrix:
   fast_finish: true
   allow_failures:
   - python: nightly
 
   include:
-  - name: 3.7 without extensions
-    python: 3.7
+  - <<: *mainstream_python_base
+    name: without extensions
     env:
-      AIOHTTP_NO_EXTENSIONS: 1
+      AIOHTTP_NO_EXTENSIONS: "1"
 
-  - <<: *_doc_base
+  - <<: *lint_base
     name: Checking docs spelling
+    before_script:
+    - pip install --upgrade -r requirements/doc-spelling.txt
     script:
     - make doc-spelling
+    after_failure: cat docs/_build/spelling/output.txt
+    addons:
+      apt:
+        packages:
+        - libenchant-dev
 
-  - <<: *_doc_base
+  - <<: *lint_base
     name: Checking Towncrier fragments
-    install:
-    - *upgrade_python_toolset
-    - pip install -r requirements/cython.txt
-    - AIOHTTP_NO_EXTENSIONS=1 pip install -r requirements/ci.txt
-    - pip install -r requirements/towncrier.txt
+    before_script:
+    - pip install --upgrade -r requirements/towncrier.txt
     script:
     - towncrier --yes
 
-  - <<: *_lint_base
+  - <<: *lint_base
     name: Linting source code with flake8
-    install:
-    - *upgrade_python_toolset
-    - pip install -r requirements/flake.txt
+    before_script:
+    - pip install --upgrade -r requirements/flake.txt
     script:
     - flake8 aiohttp examples tests
 
-  - <<: *_lint_base
+  - <<: *lint_base
     name: Linting source code with mypy
-    install:
-    - *upgrade_python_toolset
-    - pip install -r requirements/cython.txt
-    - AIOHTTP_NO_EXTENSIONS=1 pip install -r requirements/ci.txt
     script:
     - mypy aiohttp
 
-  - <<: *_lint_base
+  - <<: *lint_base
     name: Verifying distribution package metadata
-    install:
-    - *upgrade_python_toolset
-    - AIOHTTP_NO_EXTENSIONS=1 pip install -r requirements/ci.txt -r requirements/doc.txt
     script:
-    - AIOHTTP_NO_EXTENSIONS=1 python setup.py --verbose sdist bdist_wheel
+    - python setup.py --verbose sdist bdist_wheel
     - twine check dist/*
 
-  - <<: *_lint_base
+  - <<: *lint_base
     name: Making sure that CONTRIBUTORS.txt remains sorted
-    language: minimal
```

Comment:
Keep it minimal. This is to speed-up the check.

Reply:
`shell` **is** `minimal`, see the [*minimal and generic* documentation](https://docs.travis-ci.com/user/languages/minimal-and-generic/#aliases):

> Setting the `language` key to `bash`, `sh` or `shell` is equivalent to `language: minimal`.

and travis itself, when showing the effective config for a build, starts with the warning:

>  `matrix.include.language: minimal` is an alias for `shell` (value), using `shell`

Reply:
With `version: "= 0"`, Travis CI doesn't show warnings at all.

I think that `minimal` carries a better semantic meaning.

Reply:
Really? The `version` root key is virtually undocumented, and I find the travis-ci project a bit hard to navigate to figure out where this would be decided.

I so far have used https://config.travis-ci.org/ to test configurations, and it produces warnings and such regardless of the `version` key.

Reply:
Yep, totally undocumented. I stole this from some of Travis' own repos.
Anyway, I'd prefer semantically informative names. Warnings aren't errors after all. WDYT?